src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -64,8 +64,10 @@ protected function generateUrl($route, $parameters = array(), $referenceType = U
      */
     protected function forward($controller, array $path = array(), array $query = array())
     {
+        $request = $this->container->get('request_stack')->getCurrentRequest();
+        $path['_forwarded'] = $request;
```

Comment:
I don't think passing the whole parent request is a good idea. It adds a bunch of unrelated info in the subrequest attributes.